### ISSUE TITLE
Sort by date modified

### ISSUE THREAD

**Issue by Prajitura** [OxygenCobalt/Auxio#683]
### Description

Right now you can sort by date and date added, I'm proposing another option to sort by modified

### Problem solved

When I changed microSD cards and moved all music on a PC and then on the new card, musics added before the new card would all sort alphabetically, and only the ones added after the new card would sort by date added

### Other implementations

On other music players (non-foss) sorting by date modified solves this

### Benefit

You can see what musics you've added recently even if you move the files from 1 place to another, and soometimes it's easier to find a song

### Duplicates

- [X] I have checked this feature request for duplicates.
- [X] I have checked that this feature is not implemented in the [lastest version](https://github.com/OxygenCobalt/Auxio/releases).
- [X] I have checked the [Why Are These Features Missing?](https://github.com/OxygenCobalt/Auxio/wiki/Why-Are-These-Features-Missing%3F) page.
- [X] I agree to the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).

**Comment by OxygenCobalt** [comment]
Don't know about this. I really don't want to keep adding sorting options, especially since I don't see how people would use this outside of compensating for this exact issue. It seems like it would be better to merely patch Date added to handle this in some way.

**Comment by Prajitura** [comment]
Well it would help them by beig able to move files seamlessly from any device to another, without having to worry about their order. 
Does it have any disadvantages?

**Comment by OxygenCobalt** [comment]
- More cluttered sort menu
- No use outside of resolving that specific issue
- Not as useful as Date added since ex. retagging files will reorder them

**Comment by Prajitura** [comment]
Ok, will look into some batch date editors

**Comment by OxygenCobalt** [comment]
I'll be leaving this issue open for other user's feedback, as I do for others I'm unsure about.

**Comment by hellpe** [comment]
Cururiously, I was under the assumption this was already how "Date added" works. I have installed Auxio after being frustrated by VLC's inability to sort my albums by last modified, and I noticed that my Auxio library was automatically sorting my albums with the latest additions at the top. I cannot verify this since my library scan is broken until F-Droid pushes an update, but I'm surprised to learn that it doesn't use the last modification date to sort the albums.

**Comment by OxygenCobalt** [comment]
I mean, that's what "Date added" means @hellpe, right? An album shouldn't be sorted to the top if it was retagged.

**Comment by Prajitura** [comment]
Im proposing this again, When I started it I thought that it was a niche issue that happened only to me, but now realised it's impossible to move the library to another device and retain the same date-added order

**Comment by OxygenCobalt** [comment]
> Im proposing this again, When I started it I thought that it was a niche issue that happened only to me, but now realised it's impossible to move the library to another device and retain the same date-added order

This is true. But date modified also won't help unless your transfer tool preserves the timestamps between transfers. If Date added does not transfer, I doubt date modified will either.

**Comment by Prajitura** [comment]
Date added is only kept by auxio and is not an file attribute,  so there is no way to transfer it, only auxio knows the date added and it will be gone if the app is uninstalled and reinstalled somwhere else. Date modified is a file attribute respecte by every OS and program so that can be transferred through nearly all tools (I manually plug my sd card in my pc and I use syncthing)
I heard MTP has some problems with transperring date modified though

**Comment by d4f5409d** [comment]
I second this feature, I also struggle with this. I thought date added sorting meant the file's metadata not the import date to the app. As it currently is, leaves the feature useless for me and for those whom hopped applications and my song library is currently cluttered and merely confusing.

**Comment by d4f5409d** [comment]
> Im proposing this again, When I started it I thought that it was a niche issue that happened only to me, but now realised it's impossible to move the library to another device and retain the same date-added order

The thing is, issues tend to get often overlooked by developers, because most of the users don't complain. While these users may have uneasy feelings about the functionality of the app, and wish things were the other way round, they also lead to think it is not that big of a deal to complain. And I admit, it is hard to constantly wash between what to fix and leave as it is.

~Feel free to mention this anywhere you feel your issue is overlooked in free and open software development ~

**Comment by chocmake** [comment]
I can see the pros and cons. foobar2000 on Windows behaves similarly to Auxio in considering the date added to the library independent of the date modified timestamp (although in fb2k one can also set up any arbitrary sorting or grouping parameters).

For example I have releases that have old date modified timestamps but I've only recently added to foobar2000 (including from data-based music CDs I've bought which have their own timestamps). If the logic of date added were based on timestamps then this would also affect older timestamps not being considered newly added (a drawback of this approach).

The popular Mp3Tag editor allows preserving date created/modified timestamps when retagging files and is a setting I use. For libraries where the majority of music is tagged once at the point it was bought/downloaded and if the original timestamps were always preserved then ordering by date modified would represent a decent 'order I first discovered this'.

While if the files were retagged without such timestamps preservation then the order would misrepresent the sorting terminology imo, or for cases as described where existing file timestamps are desired to represent newly added files (regardless of the release date metadata).

**Comment by bengalbroys** [comment]
I third this feature, I use syncthing to transfer my library from my computer and with only date added it's impossible to keep the correct order, especially considering only Auxio knows the date added, if you reinstall the app for any reason the order is gone.

**Comment by Bsd-us** [comment]
I would love this too
